ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -123,20 +132,46 @@ class ClientQueryViewController: UITableViewController, Themeable {
 		self.extendedLayoutIncludesOpaqueBars = true
 		self.definesPresentationContext = true
 
-		sortBar = SortBar(frame: CGRect(x: 0, y: 0, width: self.tableView.frame.width, height: 40), sortMethod: sortMethod)
-		sortBar?.delegate = self
-		sortBar?.updateSortMethod()
-
-		tableView.tableHeaderView = sortBar
+		sortBar.delegate = self
+		sortBar.sortMethod = sortMethod
+		sortBar.updateSortMethod()
 
 		refreshController = UIRefreshControl()
 		refreshController?.addTarget(self, action: #selector(self.refreshQuery), for: .valueChanged)
-		self.tableView.insertSubview(refreshController!, at: 0)
-		tableView.contentOffset = CGPoint(x: 0, y: searchController!.searchBar.frame.height)
+		tableView.insertSubview(refreshController!, at: 0)
+//		tableView.contentOffset = CGPoint(x: 0, y: searchController!.searchBar.frame.height)
 
 		Theme.shared.register(client: self, applyImmediately: true)
 	}
 
+	private func render() {
+
+		sortBar.translatesAutoresizingMaskIntoConstraints = false
+		tableView.translatesAutoresizingMaskIntoConstraints = false
+
+		view.addSubview(sortBar)
+		NSLayoutConstraint.activate([
+			sortBar.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+			sortBar.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor),
+			sortBar.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor),
+			sortBar.heightAnchor.constraint(equalToConstant: 40)
+		])
+
+		view.addSubview(tableView)
+		NSLayoutConstraint.activate([
+			tableView.topAnchor.constraint(equalTo: sortBar.bottomAnchor),
+			tableView.leftAnchor.constraint(equalTo: view.leftAnchor),
+			tableView.rightAnchor.constraint(equalTo: view.rightAnchor),
+			tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+			])
+
+		tableView.delegate = self
+		tableView.dataSource = self
+
+		self.navigationController?.navigationBar.setValue(true, forKey: "hidesShadow")
```

Comment:
This is private API and shouldn't be used.

What effect does it have / why is it needed?

Reply:
The effect was to hide the line between the `NavigationBar` and the sortBar. It's only a visual effect so if this API shouldn't be used we can remove it without any problem.